src/Streamly/Internal/Data/Stream/Async/Channel/Operations.hs

```diff
@@ -129,18 +129,17 @@ fromChannelRaw sv = K.MkStream $ \st yld sng stp -> do
 
     where
 
-    allDone stp = do
-        when (svarInspectMode sv) $ do
-            t <- liftIO $ getTime Monotonic
-            liftIO $ writeIORef (svarStopTime (svarStats sv)) (Just t)
-            liftIO $ printSVar (dumpSVar sv) "SVar Done"
-        stp
+    cleanup = do
```

Comment:
This is improperly named. This will not do any `cleanup`. This is more like `loggingCleanup`.

Reply:
It is something that you do before you exit. Currently it records stats, but in general it could be anything.